SDL/LSTESData.h

```diff
@@ -0,0 +1,94 @@
+#ifndef LSTESData_H
+#define LSTESData_H
+
+#ifdef LST_IS_CMSSW_PACKAGE
+#include "RecoTracker/LSTCore/interface/alpaka/Constants.h"
+#else
+#include "Constants.h"
+#endif
+
+#include "HeterogeneousCore/AlpakaInterface/interface/CopyToDevice.h"
+
+#include <filesystem>
+#include <memory>
+
+namespace SDL {
+
+  struct pixelMap;
+
+  template <typename>
+  class TiltedGeometry;
+
+  template <typename>
+  class ModuleConnectionMap;
+  using MapPLStoLayer = std::array<std::array<ModuleConnectionMap<Dev>, 4>, 3>;
+
+  template <typename>
+  struct modulesBuffer;
+
+  template <typename>
+  class EndcapGeometryHost;
+
+  template <typename>
+  class EndcapGeometry;
+
+  template <typename TDev>
+  struct LSTESHostData;
+
+  // FIXME: This shouldn't be a templated struct
+  template <>
+  struct LSTESHostData<Dev> {
+    const std::shared_ptr<const MapPLStoLayer> mapPLStoLayer;
+    const std::shared_ptr<const EndcapGeometryHost<Dev>> endcapGeometry;
+    const std::shared_ptr<const TiltedGeometry<Dev>> tiltedGeometry;
+    const std::shared_ptr<const ModuleConnectionMap<Dev>> moduleConnectionMap;
+
+    LSTESHostData(std::shared_ptr<MapPLStoLayer> mapPLStoLayerIn,
+                  std::shared_ptr<EndcapGeometryHost<Dev>> endcapGeometryIn,
+                  std::shared_ptr<TiltedGeometry<Dev>> tiltedGeometryIn,
+                  std::shared_ptr<ModuleConnectionMap<Dev>> moduleConnectionMapIn)
+        : mapPLStoLayer(std::const_pointer_cast<const MapPLStoLayer>(mapPLStoLayerIn)),
```

Comment:
is the cast actually needed?

Reply:
https://godbolt.org/z/o8Ms81c97 from non-const to const works

as for the member constness, it's probably an overkill, since ES will share only `const LSTESHostData`

Reply:
Ah okay, I'll fix that